engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -62,6 +64,7 @@ public void initialise() {
      * @param mode a MetricsMode instance
      */
     public boolean register(MetricsMode mode) {
+        Preconditions.checkNotNull(mode, "Illegal argument passed, mode is null.");
         return modes.add(mode);
     }
```

Comment:
From the getCurrentMode() javadoc:

> Returns current mode, initializes a default {@link NullMetricsMode}, if currentMode is null

The second part of the sentence is no longer true. I'd simply write:

`* @return the current MetricsMode instance`

nothing else.